tests/pass/ptr_int_casts.rs

```diff
@@ -46,6 +46,26 @@ fn ptr_int_casts() {
 
     // involving types other than usize
     assert_eq!((-1i32) as usize as *const i32 as usize, (-1i32) as usize);
+
+    // Stress-test that the GC doesn't delete context that would prevent us from casting from int
+    // to pointer correctly after the allocation is dead.
+    let mut ptrs: Vec<*const u8> = Vec::new();
+    let mut ints: Vec<usize> = Vec::new();
+
+    for _ in 0..1_000 {
```

Comment:
Do we really need 1000 here or is 100 enough? We [run the GC after every single machine step on CI](https://github.com/rust-lang/miri/blob/f6c0b68dbbdf75b4e865f939407c142c4c6c5724/.github/workflows/ci.yml#L40) so we should be detecting issues with fairly low iteration counts, even just a single iteration in principle.

Reply:
100 is enough here. I just like powers of 1000 :p

Reply:
We don't actually run the GC after every machine step. I considered making the GC run every N machine steps, but it runs every N terminators because that's how preemption and the clock work. I'm definitely not opposed to shifting all such logic to machine steps.

Reply:
Ah, every N blocks is probably better to avoid completely grinding Miri to a halt.^^

But even then, there's a bunch of function calls here, leading to tons of blocks. So we don't even need a loop, I think?